src/Tribe/Template.php

```diff
@@ -857,10 +979,80 @@ public function template( $name, $context = [], $echo = true ) {
 		 */
 		$html = apply_filters( "tribe_template_html:$hook_name", $html, $file, $name, $this );
 
+		// Tries to hook container entry points in the HTML.
+		$html = $this->template_hook_container_entry_points( $html );
+
 		if ( $echo ) {
 			echo $html;
 		}
 
+		// Revert the current hook name.
+		$this->set_template_current_hook_name( null );
+
+		return $html;
+	}
+
+	/**
+	 * @param $html
+	 *
+	 * @return string
+	 */
+
+	/**
+	 * Run the hooks for the container entry points.
+	 *
+	 * @since  TBD
+	 *
+	 * @param string $html The html of the current template.
+	 *
+	 * @return string|false Either the final entry point content HTML or `false` if no entry point could be found or set to false.
+	 */
+	private function template_hook_container_entry_points( $html ) {
+		$regexp = '/<(?<is_end>\/)*(?<tag>[A-Z0-9]*)(?:\b)*[^>]*>/mi';
```

Comment:
Regular expression, while very powerful, are expensive operations.

The result should be instance-cached (cached for the current request duration only).

To cache something in an guaranteed ephemeral cache, use the following API:

```php
$key = md5(...);
$cache = tribe_cache();
$cached = $cache[$key];

if( false !== $cached ){
    return $cached;
}

$value = 'foo';

$cache[$key] = $value;

return $value;
```

Reply:
So, in this case, it can be complicated to cache. Because the MD5 on itself can be an expensive operation for a long string like the whole HTML. This creates a problem because the improvements are very small.

And if we use the path for caching it can fail the regular expression because the variables in the context can also dictate what will be in the first and last line, specifically for the string replacement.

Need to play around with how this behaves to see if this actually bring any performance improvements.

Reply:
I added the caching here using the html for the MD5.  So we can compare if this helps or not. We could change to use this to get the hook name:

`$hook_name = $this->get_template_current_hook_name();`

Then we could use something like this for the cache key `tribe_template_entry_point_cache:$hook_name`

Reply:
Caching on hook name sounds useful and makes sense to me.